timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -17,16 +17,27 @@
 logger = logging.getLogger(__name__)
 
 
-@dataclass
-class CovariateMetadata:
+class CovariateMetadata(BaseModel):
```

Comment:
It can still complicate things due to dependency version conflicts or just one more thing to maintain. I would rather not introduce a dependency for this sort of functionality, but your call.

Reply:
Fair enough, I guess this can be handled by a static type checker without introducing a dependency